compiler/circle-mpqsolver/src/bisection/Evaluator.cpp

```diff
@@ -0,0 +1,155 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Evaluator.h"
+
+#include <luci_interpreter/Interpreter.h>
+
+#include <dio_hdf5/HDF5Importer.h>
+
+#include <cmath>
+#include <iostream>
+
+using namespace mpqsolver::bisection;
+
+using Shape = std::vector<loco::Dimension>;
+
+namespace
+{
+
+using namespace luci;
+
+template <typename NodeT> size_t get_tensor_size(const NodeT *node)
+{
+  uint32_t tensor_size = loco::size(node->dtype());
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    tensor_size *= node->dim(i).value();
+  return tensor_size;
+}
+
+DatasetOutput compute_outputs(const luci::Module *module, const std::string &h5file)
+{
+  dio::hdf5::HDF5Importer importer{h5file};
+  importer.importGroup("value");
+
+  bool is_raw_data = importer.isRawData();
+
+  const auto num_records = importer.numData();
+  if (num_records == 0)
+    throw std::runtime_error("The input data file does not contain any record.");
+  const auto input_nodes = loco::input_nodes(module->graph());
+  const auto num_inputs = input_nodes.size();
+
+  DatasetOutput dataset_output;
+
+  // Create interpreter.
+  luci_interpreter::Interpreter interpreter(module);
+  for (int32_t record_idx = 0; record_idx < num_records; record_idx++)
+  {
+    if (num_inputs != importer.numInputs(record_idx))
+      throw std::runtime_error("Wrong number of inputs.");
+    for (int32_t input_idx = 0; input_idx < num_inputs; input_idx++)
+    {
+      const auto *input_node = loco::must_cast<const luci::CircleInput *>(input_nodes[input_idx]);
+      assert(input_node->index() == input_idx);
+
+      std::vector<char> input_data(get_tensor_size(input_node));
+
+      if (!is_raw_data)
+      {
+        loco::DataType dtype;
+        Shape shape;
+        importer.readTensor(record_idx, input_idx, &dtype, &shape, input_data.data());
+      }
+      else
+      {
+        // Skip type/shape check for raw data
+        importer.readTensor(record_idx, input_idx, input_data.data());
+      }
+
+      interpreter.writeInputTensor(input_node, input_data.data(), input_data.size());
+    }
+
+    interpreter.interpret();
+
+    NNOutput nn_output;
+
+    // Get output.
+    const auto output_nodes = loco::output_nodes(module->graph());
+    for (int i = 0; i < module->graph()->outputs()->size(); i++)
+    {
+      const auto *output_node = loco::must_cast<const luci::CircleOutput *>(output_nodes[i]);
```

Comment:
Please check output_node's dtype is supported type (float32 as of now).

Reply:
ok